Sail/src/API/DX12/dxr/DXRBase.cpp

```diff
@@ -394,20 +403,23 @@ void DXRBase::reloadShaders() {
 	createRaytracingPSO();
 }
 
-bool DXRBase::onEvent(Event& event) {
-	auto onResize = [&](WindowResizeEvent& event) {
+bool DXRBase::onEvent(const Event& event) {
+	auto onResize = [&](const WindowResizeEvent& event) {
 		// Window changed size, resize output UAV
 		createInitialShaderResources(true);
 		return true;
 	};
 
-	auto onGameOver = [&](GameOverEvent& event) {
```

Comment:
This looks suspicious, the fact that 'onGameOver' is completely removed and not replaced with a modified version. If this is intended, ignore this comment.

Reply:
DXRBase does not care about when the game is over, it just wants to know when to reset the water.
Game over event replaced by reset water event in this file.